lib/dns.js

```diff
@@ -289,31 +289,38 @@ function setServers(servers) {
   // servers cares won't have any servers available for resolution
   const orig = this._handle.getServers();
   const newSet = [];
-  const IPv6RE = /\[(.*)\]/;
   const addrSplitRE = /(^.+?)(?::(\d+))?$/;
 
   servers.forEach((serv) => {
+    // Easily recognized?
     var ipVersion = isIP(serv);
     if (ipVersion !== 0)
       return newSet.push([ipVersion, serv, IANA_DNS_PORT]);
 
-    const match = serv.match(IPv6RE);
-    // we have an IPv6 in brackets
-    if (match) {
-      ipVersion = isIP(match[1]);
+    // IPv6: /\[(.*)\]/ but O(n) complexity
+    const openBracketIx = serv.indexOf('[');
+    const nextCloseBracketIx = serv.indexOf(']', openBracketIx);
+    if (0 <= openBracketIx && openBracketIx < nextCloseBracketIx) {
```

Comment:
Suggestion: change it to e.g.,

```js
const openBracketIx = serv.indexOf('[');
if (openBracketIx !== -1) {
  const nextCloseBracketIx = serv.indexOf(']', openBracketIx);
  if (nextCloseBracketIx > openBracketIx) {
    const withinBrackets = ...
    ...
  }
}
```

Reply:
Neither. It is just faster because it will not check for the closing bracket in case the first one does not exist.

Reply:
It also looks more Obviously Correct.

Apropos the names, can you use `fooIndex` instead of `fooIx`?